packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,370 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:vm_snapshot_analysis/treemap.dart';
+
+import '../base/file_system.dart';
+import '../convert.dart';
+import 'logger.dart';
+import 'process.dart';
+import 'terminal.dart';
+
+/// A class to analyze APK and AOT snapshot and generate a breakdown of the data.
+class SizeAnalyzer {
+  SizeAnalyzer({
+    @required this.fileSystem,
+    @required this.logger,
+    @required this.processUtils,
+  });
+
+  final FileSystem fileSystem;
+  final Logger logger;
+  final ProcessUtils processUtils;
+
+  static const String aotSnapshotFileName = 'aot-snapshot.json';
+
+  static const int tableWidth = 80;
+
+  /// Analyzes [apk] and [aotSnapshot] to output a [Map] object that includes
+  /// the breakdown of the both files, where the breakdown of [aotSnapshot] is placed
+  /// under 'lib/arm64-v8a/libapp.so'.
+  ///
+  /// The [aotSnapshot] can be either instruction sizes snapshot or v8 snapshot.
+  Future<Map<String, dynamic>> analyzeApkSizeAndAotSnapshot({
+    @required File apk,
+    @required File aotSnapshot,
+  }) async {
+    logger.printStatus('▒' * tableWidth);
+    _printEntitySize(
+      '${apk.basename} (total compressed)',
+      byteSize: apk.lengthSync(),
+      level: 0,
+      showColor: false,
+    );
+    logger.printStatus('━' * tableWidth);
+    final Directory tempApkContent = fileSystem.systemTempDirectory.createTempSync('flutter_tools.');
+    // TODO(peterdjlee): Implement a way to unzip the APK for Windows. See issue #62603.
+    String unzipOut;
+    try {
+      // TODO(peterdjlee): Use zipinfo instead of unzip.
```

Comment:
FYI, as far as I know we do not document requiring zipinfo to be installed, so switching may not be straightforward

Reply:
both zipinfo and unzip are built-in on macOS and neither are built-in on Ubuntu so we're more or less in the same position either way.

Reply:
No, we're not in the same state - we require unzip and warn if it is not installed and do no such thing for zipinfo

Reply:
ah I see, SG, let's leave it then. This makes it simpler.